experimental/dispatch_profiler/README.md

```diff
@@ -0,0 +1,173 @@
+# IREE Dispatch Profiler
+
+The IREE Dispatch Profiler is a Python-based tool designed to achieve two primary objectives: functional verification and performance profiling for individual dispatches, such as matrix multiplication, batch matrix multiplication, and convolutions. This tool ensures that performance optimizations maintain functionality and provides a convenient way to quantitatively measure performance. Additionally, the tool offers dispatch generation and compilation capabilities. In summary, the IREE dispatch profiler accomplishes the following:
```

Comment:
Nit: for a future PR. please wrap markdown source at 80 columns. For VSCode I use the [reflow-markdown](https://marketplace.visualstudio.com/items?itemName=marvhen.reflow-markdown) extension (you still have to do each paragraph separately.